core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1716,7 +1716,7 @@ class KafkaController(val config: KafkaConfig,
       case Some(replicas) =>
         val replicaSet = replicas.toSet
 
-        if (replicas.size != replicaSet.size)
+        if (replicas.isEmpty || replicas.size != replicaSet.size)
```

Comment:
Thanks @stanislavkozlovski . I think this only gets called for the API path. Should we perform a similar check for the ZK side? This is around line 1698 in this file.